packages/backend/src/models/SpaceModel.ts

```diff
@@ -1532,4 +1648,49 @@ export class SpaceModel {
             .andWhere('group_uuid', groupUuid)
             .delete();
     }
+
+    /**
+     * Gets the root space UUID for a given space UUID
+     *
+     * This method uses PostgreSQL's ltree extension to find the root space of a hierarchy.
+     * The spaces are stored in a tree structure where:
+     * - Root spaces have a path with a single level (e.g., "my-space")
+     * - Child spaces have paths that include their parent hierarchy (e.g., "my-space.my-child-space")
+     * @param spaceUuid Space UUID to get the root for
+     * @returns Root space UUID (or the same UUID if it's already a root or has no hierarchy)
+     */
+    async getSpaceRoot(spaceUuid: string): Promise<string | undefined> {
+        const result = await this.database(SpaceTableName)
+            .select<{ root_space_uuid: string; is_self_root: boolean }>(
+                this.database.raw(`
+                CASE 
+                    WHEN path IS NOT NULL THEN 
+                        (SELECT root_space.space_uuid 
+                        FROM spaces root_space 
+                        -- Using subpath(path, 0, 1) to extract the first element of the path
+                        -- This gets just the root part of the path
+                        WHERE subpath(root_space.path, 0, 1) = subpath(${SpaceTableName}.path, 0, 1)
+                        -- nlevel(path) = 1 finds only root nodes (first level in hierarchy)
+                        AND nlevel(root_space.path) = 1)
+                    ELSE space_uuid 
+                END as root_space_uuid,
+                -- Check if this space is its own root (either it's a root space or has no hierarchy)
+                CASE 
+                    WHEN path IS NOT NULL THEN 
+                        (nlevel(path) = 1 OR space_uuid = (
+                            SELECT root_space.space_uuid 
+                            FROM spaces root_space 
+                            WHERE subpath(root_space.path, 0, 1) = subpath(${SpaceTableName}.path, 0, 1)
+                            AND nlevel(root_space.path) = 1
+                        ))
+                    ELSE true
+                END as is_self_root
+            `),
+            )
+            .where('space_uuid', spaceUuid)
+            .first();
+
+        // Return undefined if the space is its own root
+        return result?.is_self_root ? undefined : result?.root_space_uuid;
```

Comment:
What's the reason for this? I'd expect same `spaceUuid` to be returned if passed` spaceUuid` is "root" (top level)

Reply:
taken care of in #14312